### PR TITLE
Build project with CameraState

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### CODE REVIEW

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt [hunk 1]
```diff
@@ -51,7 +51,7 @@ class CameraAnimatorsFactory internal constructor(mapDelegateProvider: MapDelega
       )
     }
     cameraOptions.bearing?.let {
-      var startBearing = currentCameraOptions.bearing ?: it
+      var startBearing = currentCameraState.bearing ?: it
```
Comment (macdrevx) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt:54]: `?: it` shouldn't be needed anymore, right?

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt [hunk 2]
```diff
@@ -133,7 +126,7 @@ class CameraAnimatorsFactory internal constructor(mapDelegateProvider: MapDelega
    * @return Array of the created animators
    */
   internal fun getPitchBy(pitch: Double): Array<CameraAnimator<*>> {
-    mapCameraDelegate.getCameraOptions().pitch?.let { startPitch ->
+    mapCameraDelegate.getCameraState().pitch.let { startPitch ->
```
Comment (macdrevx) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt:129]: `.let` shouldn't be needed anymore, right?

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt (hunk 1 above)
Comment (kiryldz) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt:54]: True! Thanks for catching!

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt (hunk 2 above)
Comment (kiryldz) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt:129]: Yep, using `.let` and `?.let` is also fine but we could get rid of it in this case.
Thanks 👍

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt [hunk 3]
```diff
@@ -275,26 +260,14 @@ class CameraAnimatorsFactory internal constructor(mapDelegateProvider: MapDelega
...
-    var endZoom = cameraOptions.zoom
-      ?: currentCameraOptions.zoom
-      ?: return emptyArray()
-    var endBearing = cameraOptions.bearing
-      ?: currentCameraOptions.bearing
-      ?: return emptyArray()
-    val endPitch = cameraOptions.pitch
-      ?: currentCameraOptions.pitch
-      ?: return emptyArray()
+    val currentCameraOptions = mapCameraDelegate.getCameraState()
```
Comment (macdrevx) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimatorsFactory.kt:263]:
```suggestion
    val currentCameraState = mapCameraDelegate.getCameraState()
```

**Comment by kiryldz** [comment]
Closing, following up in https://github.com/mapbox/mapbox-maps-android/pull/304

**Pull request closed without merging**